scripts/buildtools/check_tests.py

```diff
@@ -0,0 +1,116 @@
+print("===== STARTED RUNNING check_tests.py =====\n")
+
+import json
+import sys
+import argparse
+
+# Arguments parsing / help menu
+parser = argparse.ArgumentParser(description='Check test results for intentionally and unintentionally skipped tests, as well as tests that ran multiple times.')
+parser.add_argument('tests_results_filepath', metavar='RESULTS_FILE',
+    help='json format test results file path (e.g. /tmp/results/testresults.json)')
+args = parser.parse_args()
+
+# Go through the given file one json object at a time, and record into a dict
+AllTestResults = {}
+with open(args.tests_results_filepath) as f:
+    for jsonObj in f:
+        testDict = json.loads(jsonObj)
+        if 'Test' not in testDict:
+            continue
+        
+        fullTestName = testDict['Package'] + ' ' + testDict['Test']
+        if fullTestName not in AllTestResults:
+            AllTestResults[fullTestName] = {}
+            AllTestResults[fullTestName]['ran'] = 0
+            AllTestResults[fullTestName]['last_output'] = ''
+            AllTestResults[fullTestName]['skipped_intentionally'] = False
+            AllTestResults[fullTestName]['skipped_reason'] = ''
+
+        # actions can be: output, run, skip, pass
+        if 'pass' in testDict["Action"]:
+            AllTestResults[fullTestName]['ran'] += 1
+        elif 'output' in testDict["Action"]:
+            if '--- SKIP' in testDict['Output'] and 'due to partitioning' not in AllTestResults[fullTestName]['last_output']:
+                AllTestResults[fullTestName]['skipped_intentionally'] = True
+                if '=== RUN' in AllTestResults[fullTestName]['last_output']:
+                    AllTestResults[fullTestName]['skipped_reason'] = 'No reason given. PLEASE CHECK!!!'
+                else:
+                    AllTestResults[fullTestName]['skipped_reason'] = AllTestResults[fullTestName]['last_output'].strip()
+
+            AllTestResults[fullTestName]['last_output'] = testDict['Output']
+
+            
+
+f.close()
```

Comment:
I don't think `f.close()` is necessary if everything else is run within the `with open(args.tests_results_filepath) as f`

Reply:
good point, but it doesn't hurt anything, so we might want to leave it for now, just to finish the ticket.

Reply:
just because of scope creep already, we are already cleaning up a lot of tests, which is not part of this ticket technically.